src/hooks/api.tsx

```diff
@@ -0,0 +1,61 @@
+import axios from 'axios';
+
+const END_POINT = 'http://kdt.frontend.3rd.programmers.co.kr:5006';
+
+interface Iprops {
+  url: string;
+  method: string;
+  body?: object;
+  headers?: object;
+}
+
+export const useCustomApi = async ({ url, body, headers, method }: Iprops) => {
+  // const token = localStorage.getItem('TOKEN_KEY');
+  // if (!token) return console.warn('Need Token');
+
+  switch (method) {
+    case 'GET':
+      try {
+        return await axios
+          .get(`${END_POINT}${url}`, {
+            data: {
+              ...body,
+              ...headers,
+            },
+          })
```

Comment:
ask;
혹시 data에 headers를 넣어주신 이유가 있나요? 제가 알고 있기로는 data는 body로 전달 될 값이 들어간다고 알고있어서요. 그리고 현재 저희 API 설계서 상에는 GET 요청에서 body 대신 params 데이터를 넘기는게 어떨까 제안해봅니다.

```suggestion
          .get(`${END_POINT}${url}`, {
                ...params,
        	...headers,
          })
```

Reply:
get 요청의 경우 2개의 파라미터가 input으로 기되된다고 나오더라구요. 
1개는 url 이고 다른 하나만 더 받을 수 있었습니다. 
그래서 url, body, headers의 형태로 요청을 하게 되면 에러가 발생하더라구요.

params라 하면 useParams를 사용하여 파라미터를 받아오는 것을 말씀하시는 걸까요?

Reply:
네 보니까 params가 필요한 api는 4가지 정도 밖에 되지 않네요. body를 통해 params 정보를 넘겨주는 방법도 좋다고 생각하는데 get 요청에서 body를 넘겨줘도 괜찮은지 모르겠습니다.(저도 이 부분에 있어 제대로 된 지식이 없다보니...)

Reply:
get 요청은 제가 알기로 데이터를 요청해서 받아오는 것이기 때문에 body를 넣을 수는 있지만 굳이 사용하지 않는 방법인 거 같아요! 대신 get 요청 시 어떤 데이터(e.g. 페이지네이션 시 인덱스 처럼)를 보내야 한다면 params로 넘겨준다고 알고 있습니다ㅎㅎ